hazelcast/test/src/compact/compact_schema_replication_on_cluster_restart_test.h

```diff
@@ -54,6 +79,20 @@ TEST_F(CompactSchemaReplicationOnClusterRestart, on_cluster_restart)
                            check_schema_on_backend(schema_child));
 }
 
+TEST_F(CompactSchemaReplicationOnClusterRestart,
+       dont_replicate_if_there_are_no_schema_on_cluster_restart)
+{
+    auto condition = [this]() { return number_of_schemas_on_member() == 0; };
+
+    ASSERT_TRUE_EVENTUALLY(condition());
+
+    member_.shutdown();
+
+    HazelcastServer another_member{ factory_ };
+
+    ASSERT_TRUE_EVENTUALLY(condition());
```

Comment:
First make sure that client is connected back and then use `ASSERT_TRUE_ALL_THE_TIME` and wait for 1 sec.

Reply:
Fixed.